Samples/WindowPlacement/README.md

```diff
@@ -0,0 +1,858 @@
+   not think of it as a change in coordinate space).
+
+   So:
+
+    - If a monitor exists with the same name, use that monitor.
+
+    - If that's not available, fall back to `MonitorFromRect` with
+      `MONITOR_DEFAULTTONEAREST`.
+
+2. **Adjust the normal rect** (and Arrange rect if Arranged) for this desired
+   monitor.
+
+    - Each rect (normal/arranged) is transformed in a different way.
+
+    - See previous section.
+
+3. **Disable painting & animations**, if needed.
+
+    - In some cases, it will be necessary to move the window multiple times. To
+      avoid flickering, temporarily disable painting and animations.
+
+    - Calling `WM_SETREDRAW` with `false` for `wParam` disables painting for the
+      window. While disabled, the window's contents on screen will not be
+      changed (though the window will not be hidden, if it is visible). When
+      re-enabling painting (`WM_SETREDRAW` `true`) at the end, the window will need
+      to be explicitly invalidated (repainted) and activated, since both are
+      skipped while `WM_SETREDRAW` false.
+
+    - `SystemParametersInfo`'s `SPI_SETANIMATION` sets a global user setting for
+      animating window transitions (Max/Min/Arrange/Restore). If moving the
+      window multiple times, it is best to temporarily disable animations. This
+      ensures the window isn't shown animating from a position the user didn't
+      see the window at.
+
+      ```c++
+      DefWindowProc(hwnd, WM_SETREDRAW, 1, 0);
+
+      ANIMATIONINFO animationInfo = { sizeof(ANIMATIONINFO) };
+      SystemParametersInfo(SPI_GETANIMATION, 0, &animationInfo, 0);
+      bool needsAnimationReset = !!animationInfo.iMinAnimate;
+      if (needsAnimationReset)
+      {
+         animationInfo.iMinAnimate = false;
+         SystemParametersInfo(SPI_SETANIMATION, 0, &animationInfo, 0);
+      }
+
+      // ... move window multiple times
+
+      DefWindowProc(hwnd, WM_SETREDRAW, 1, 0);
+      SetActiveWindow(hwnd);
+      InvalidateRect(hwnd, nullptr, true);
+
+      if (needsAnimationReset)
+      {
+         animationInfo.iMinAnimate = true;
+         SystemParametersInfo(SPI_SETANIMATION, 0, &animationInfo, 0);
+      }
+      ```
+
+4. **Restore the window**. If a window is Maximized, Minimized, or Arranged, the
+      window needs to be restored (SW_RESTORE) before it is moved to the desired
+      normal position.
+
+    - The normal position is defined as the last position the window had before
+         Maximizing/Minimizing/Arranging.
+
+    - If Maximized and moving to another monitor (but staying Maximized), the
+         window must be restored first, moved to the other monitor, and then
+         Maximized. Moving only once would leave the normal position on the
+         previous monitor (restoring the window would move it to another monitor).
+
+5. **Move the window to the correct monitor**. If a window is changing monitors, it must be moved to that monitor prior to
```

Comment:
Some notes:

The point of these steps is that we must move the window to the correct monitor; doing this correctly means _restoring_ the window, moving it to its new monitor, and then reapplying any necessary max/min/arrange state. This guarantees that the window is in the correct state AND has a correctly-set restore position.

Also: `SetWindowPlacement` _technically_ does a lot of this for you. But it doesn't handle multiple monitors. It will also unarrange you. 

So all this is "Moving a window to a stored position", including the normal rect, arrange state, activation, monitor, etc.